core/src/main/java/com/linecorp/armeria/client/AbstractEventLoopEntry.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import com.linecorp.armeria.common.util.ReleasableHolder;
+
+import io.netty.channel.EventLoop;
+
+abstract class AbstractEventLoopEntry implements ReleasableHolder<EventLoop> {
+
+    private final AbstractEventLoopState parent;
+    private final EventLoop eventLoop;
+
+    AbstractEventLoopEntry(AbstractEventLoopState parent, EventLoop eventLoop) {
+        this.parent = parent;
+        this.eventLoop = eventLoop;
+    }
+
+    @Override
+    public EventLoop get() {
+        return eventLoop;
+    }
+
+    @Override
+    public void release() {
+        parent.release(this);
+    }
+
+    abstract int activeRequests();
```

Comment:
Curious are these/below methods needed to treat as abstract methods? I think it's not invoked outside of state ?

Reply:
`decrementActiveRequests` and `index` are used in another [class](https://github.com/line/armeria/pull/1886/files/0e69530c861059c85addd0dabf2eed163bed6012#diff-3ca01fccde269ac03030409d2526aa77R102)
and `activeRequests` and `id` are used in the test cases.
We can hide `incrementActiveRequests` and `setIndex` but I thought exposing `decrementActiveRequests` and hiding `incrementActiveRequests` is kinna weird.
So there's only `setIndex` left. To hide it I had to change `List<AbstractEventLoopEntry> entries` to `List<? extends AbstractEventLoopEntry> entries` and changed related things.
But this is package-private so I thought it's not worth it. 😄

Reply:
If some methods are only used in tests, let's annotate if `@VisibleForTesting`.

Reply:
> If some methods are only used in tests, let's annotate if @VisibleForTesting.

They are used in `AbstractEventLoopState` as well :)
I was talking about when I Changed to using `List<? extends AbstractEventLoopEntry> entries` in `AbstractEventLoopState` and adding `Entry` to the list. In that case, I have to annotate them with `@VisibleForTesting`. But I decided to add `AbstractEventLoopEntry` to the list, so I don't have to. :-)